src/mlpack/methods/ann/layer/hardshrink.hpp

```diff
@@ -0,0 +1,211 @@
+/**
+ * @file hardshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * Same as soft thresholding, if its amplitude is smaller than a predefined
+ * threshold, it will be set to
+ * zero (kill), otherwise it will be kept unchanged.
+ * In order to promote sparsity and to improve the approximation, the hard
+ * thresholding method is used as an alternative.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_HARDSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_HARDSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * HardShrink function is defined as,
+ *         | x , if x >  lambda
+ *  f(x) = | x , if x < -lambda
+ *         | 0, otherwise
+ *
+ * lambda is set to 0.5 by default.
+ *
+ *          | 1 , if x >  lambda
+ *  f'(x) = | 1 , if x < -lambda
+ *          | 0 , otherwise
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class HardShrink
+{
+ public:
+  /**
+   * Create HardShrink object using specified hyperparameter lambda.
+   * Hard shrinkage has a threshold lambda which is calculated by multiplying the
+   * noise level sigma of the input(noisy image) and a coefficient 'a' which is one
+   * of the training parameter.
+   * Default (lambda = 0.5). lambda should be > 0.
+   * 
+   * @param lambda parameter
```

Comment:
Can you elaborate on the parameter?

Reply:
Hi, I have tried to do the same in new changes.
Kindly see if the are up to the mark.
Thanks.